### ISSUE TITLE
Download of trial tool fails - Reports "Damaged" by OSX

### ISSUE THREAD

**Issue by marvingreenberg** [fifafu/BetterTouchTool#2122]
**IMPORTANT:** Don't post private requests here (e.g. license issues, etc.). Please send me an email at boastr.net@gmail.com instead.

---

#### Description of bug/feature request/question:
I downloaded the BetterTouchTool.  Immediately, OSX ArchiveUtility reported the zip file was damaged.  I assumed this was simple Apple being annoying, unzipped the download manually - it seemed perfectly fine. 

MD5 (/Users/mgreenberg/Downloads/BetterTouchTool.zip) = 8dfa3e1563611925bd6d8b3843d62f0b
Code signing seems valid per instructions, but trying to run the app fails, see image.

#### Device information:

Type of Mac: iMac Pro (2017) (ooh ooh)
macOS version: 10.13.13
BetterTouchTool version: latest

#### Additional information (e.g. StackTraces, related issues, screenshots, workarounds, etc.):

<img width="1007" alt="bettertouch_fail" src="https://user-images.githubusercontent.com/216707/36313805-d0aebeec-1300-11e8-8795-9c7eebb04242.png">

---

**Note:** Before bug reporting, please make sure you have the latest version of BetterTouchTool and that you have already tried to restart your system :-). If you encounter a crash, please attach a crash log from the macOS Console.app from the "User Diagnostic Reports" section.

**Comment by fifafu** [comment]
Have you tried to re-download it?

**Comment by marvingreenberg** [comment]
Obviously?  But you could also see if the md5 sum is what you expect.

    MD5 (/Users/mgreenberg/Downloads/BetterTouchTool.zip) = 8dfa3e1563611925bd6d8b3843d62f0b

**Comment by fifafu** [comment]
Jup that's the correct hash (also it's working fine here).

Could you try to restart your Mac?

**Comment by fifafu** [comment]
Were you able to get it to run?
Maybe also try to move it to your applications folder first.

**Comment by marvingreenberg** [comment]
I just tried it here at home, and it worked fine, without moving it to Applications or anything.  I would say that maybe my company's firewall was messing with a "zip" file somehow, but you say the checksum is correct (suggestion: put that somewhere on the web site...)

I'll try again tomorrow, see if anything has changed.

Thanks.

**Comment by fifafu** [comment]
Yep it's really weird that the checksum matches - because of this I don't think the zip file is damaged. It could be the macOS code signature service crashed or hung up.

Would be great if you'd let me know in case you figure out what caused this.

**Comment by marvingreenberg** [comment]
Maybe this will give you some clues.  I rebooted.  Tried again to unzip - warned that the zip file was corrupted (a lie, probably OSX trying to prevent from unzipping the ".app" it doesn't like.)  Unzip from command line, move the app to /Applications.  Try to run get same error above.  Try to run the executable explicitly, now it says something different.

<img width="602" alt="bttrelaunch" src="https://user-images.githubusercontent.com/216707/36484844-8ad25870-16e8-11e8-9b5f-c7fd773fffec.png">

But BTT actually starts now, displaying the message about needed to use the Accessibility API, and displaying the icon on the menu bar.   Interestingly, going to the System Settings, BTT is not displayed as an App to grant access to Accesibility (because Relaunch failed?)
From the BTT menu, I try to run Relaunch (since that seems related to the issue) and a variety of errors are logged on the Terminal window where I started BTT.

<img width="607" alt="relauncherrors" src="https://user-images.githubusercontent.com/216707/36485028-15df920c-16e9-11e8-8021-273e3551c73c.png">

And finally I tried to just manually run relaunch itself,

    /Applications/BetterTouchTool.app/Contents/Resources/relaunch 
    /BuildRoot/Library/Caches/com.apple.xbs/Sources/AppleFSCompression/AppleFSCompression-96.30.2/Common/ChunkCompression.cpp:50: Error: unsupported compressor 8
    /BuildRoot/Library/Caches/com.apple.xbs/Sources/AppleFSCompression/AppleFSCompression-96.30.2/Libraries/CompressData/CompressData.c:353: Error: Unknown compression scheme encountered for file 
    '/System/Library/CoreServices/CoreTypes.bundle/Contents/Resources/Exceptions.plist'
    /BuildRoot/Library/Caches/com.apple.xbs/Sources/AppleFSCompression/AppleFSCompression-96.30.2/Common/ChunkCompression.cpp:50: Error: unsupported compressor 8
    /BuildRoot/Library/Caches/com.apple.xbs/Sources/AppleFSCompression/AppleFSCompression-96.30.2/Libraries/CompressData/CompressData.c:353: 
    Error: Unknown compression scheme encountered for file 
     '/System/Library/CoreServices/CoreTypes.bundle/Contents/Library/AppExceptions.bundle/Exceptions.plist'
    2018-02-21 09:26:46.327 relaunch[1233:92614] *** Terminating app due to uncaught exception 
    'NSInvalidArgumentException', reason: '*** +[NSString stringWithCString:encoding:]: NULL cString' 
    *** Call stack at first throw:
    (
	0   CoreFoundation                      0x93a5dfc2 __raiseError + 178
	1   libobjc.A.dylib                     0xa6af5346 objc_exception_throw + 273
	2   CoreFoundation                      0x93a5dee5 +[NSException raise:format:] + 101
	3   Foundation                          0x9531dea1 +[NSString stringWithCString:encoding:] + 79
	4   relaunch                            0x00001dbd main + 183
	5   relaunch                            0x00001cda start + 54
	6   ???                                 0x00000001 0x0 + 1
    )

Maybe that will all give you some clues or things to check.  This Mac is a new 2017 iMac, that was just installed from a backup of a 4 year old MacPro.  I guess I don't need BTT.

**Comment by fifafu** [comment]
Super weird.

One more question:
is it possible you or your company have enabled the "Only From App Store" option in System Preferences (see screenshot):

**Comment by marvingreenberg** [comment]
Well. That is true that it was set to 'App Store' only.  But switching it to '... and identified developers' didn't change anything.  Meanwhile, other apps Like Emacs, Docker, Chrome all work fine.  I mean, who downloads apps from the AppStore?

You can close the case if you like.  I was only marginally interested in BTT.

**Comment by fifafu** [comment]
Just out of interest: did you install these other apps before restoring the backup or after?
(macOS whitelists apps that have been started once. Additionally I think changing the setting to "App Store and identified developers" will require a restart in some cases)

I'm closing here because this is not a bug in BTT. However the error message definitely is stupid.

**Comment by marvingreenberg** [comment]
This is the issue, some OSX problem with migration assistant (which I used).  I can download a ".jpg" from the internet and see this behavior

https://discussions.apple.com/thread/8091630
